daemon/daemon_main.go

```diff
@@ -1407,7 +1415,38 @@ func (d *Daemon) instantiateAPI() *restapi.CiliumAPI {
 }
 
 func initKubeProxyFreeOptions() {
+	if option.Config.KubeProxyReplacement != option.KubeProxyReplacementStrict &&
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementPartial &&
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementProbe {
+		log.Fatalf("Invalid value for --%s: %s", option.KubeProxyReplacement, option.Config.KubeProxyReplacement)
+	}
+
+	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementProbe {
+		option.Config.EnableNodePort = true
+		option.Config.EnableExternalIPs = true
+		option.Config.EnableHostReachableServices = true
+		option.Config.EnableHostServicesTCP = true
+		option.Config.EnableHostServicesUDP = true
+	}
+
+	if option.Config.EnableNodePort && option.Config.EnableIPSec {
```

Comment:
I think a `switch case` would make this a little more readable .
```
switch option.Config.KubeProxyReplacement {
   case option.KubeProxyReplacementProbe:
    ...
    default:
    log.Fatalf...
}
```

Reply:
Wouldn't that lead to a lot of code duplication?